tests/mappers/test_exon_genomic_coords.py

```diff
@@ -19,286 +20,533 @@ def test_egc_mapper(test_cool_seq_tool):
+                },
+                "end": 156881850,
+            },
+        },
     }
-    return GenomicData(**params)
+    return GenomicTxSegService(**params)
 
 
 @pytest.fixture(scope="module")
 def zbtb10_exon3_end():
-    """Create test fixture for ZBTB10, end of exon 3"""
+    """Create test fixture for ZBTB10, end of exon 3 (positive strand)"""
     params = {
         "gene": "ZBTB10",
-        "chr": "NC_000008.11",
-        "start": None,
-        "end": 80514009,
-        "exon_start": None,
-        "exon_end": 3,
-        "exon_end_offset": 100,
-        "exon_start_offset": None,
-        "transcript": "NM_001105539.3",
-        "strand": Strand.POSITIVE,
+        "genomic_accession": "NC_000008.11",
+        "tx_ac": "NM_001105539.3",
+        "seg_start": None,
+        "seg_end": {
+            "exon_ord": 2,
+            "offset": 2,
+            "genomic_location": {
+                "type": "SequenceLocation",
+                "sequenceReference": {
+                    "type": "SequenceReference",
+                    "refgetAccession": "SQ.209Z7zJ-mFypBEWLk4rNC6S_OxY5p7bs",
+                },
+                "end": 80514010,
+            },
+        },
     }
-    return GenomicData(**params)
+    return GenomicTxSegService(**params)
 
 
 @pytest.fixture(scope="module")
 def zbtb10_exon5_start():
-    """Create test fixture for ZBTB10, start of exon 5"""
+    """Create test fixture for ZBTB10, start of exon 5 (positive strand)"""
     params = {
         "gene": "ZBTB10",
-        "chr": "NC_000008.11",
-        "start": 80518580,
-        "end": None,
-        "exon_start": 5,
-        "exon_start_offset": -374,
-        "exon_end": None,
-        "exon_end_offset": None,
-        "transcript": "NM_001105539.3",
-        "strand": Strand.POSITIVE,
+        "genomic_accession": "NC_000008.11",
+        "tx_ac": "NM_001105539.3",
+        "seg_start": {
+            "exon_ord": 4,
+            "offset": -201,
+            "genomic_location": {
+                "type": "SequenceLocation",
+                "sequenceReference": {
+                    "type": "SequenceReference",
+                    "refgetAccession": "SQ.209Z7zJ-mFypBEWLk4rNC6S_OxY5p7bs",
+                },
+                "start": 80518580,
+            },
+        },
+        "seg_end": None,
     }
-    return GenomicData(**params)
+    return GenomicTxSegService(**params)
 
 
 @pytest.fixture(scope="module")
 def tpm3_exon6_end():
-    """Create test fixture for TPM3, end of exon 6"""
+    """Create test fixture for TPM3, end of exon 6 (negative strand)"""
     params = {
         "gene": "TPM3",
-        "chr": "NC_000001.11",
-        "start": None,
-        "end": 154171409,
-        "exon_start": None,
-        "exon_start_offset": None,
-        "exon_end": 6,
-        "exon_end_offset": 3,
-        "transcript": "NM_152263.4",
-        "strand": Strand.NEGATIVE,
+        "genomic_accession": "NC_000001.11",
+        "tx_ac": "NM_152263.4",
```

Comment:
should we remain consistent with which transcript version we use? Here it's version 4, but above it was version 3.

Reply:
.4 is used here because no transcript was provided, so MANE transcript is selected.